apps/src/templates/HintsDisplay.jsx

```diff
@@ -31,6 +33,30 @@ module.exports = React.createClass({
     });
   },
 
+  renderBlocklyHint: function (hint) {
+    var node = document.getElementById(hint.hintId);
+    // Only render if the node exists in the DOM
+    if (node) {
+      Blockly.BlockSpace.createReadOnlyBlockSpace(node, hint.block);
+    }
```

Comment:
Absolutely agreed; the problem is that this React component now contains Blockly as a dependency, which has some serious issues with rendering outside the DOM. I briefly looked into fixing that problem of Blockly's, but it looks like it would require some pretty significant refactoring.